tests/test_generic.py

```diff
@@ -972,32 +972,32 @@ def test_create_string_object_force():
 @pytest.mark.parametrize(
     ("value", "expected"),
     [
-        ("0.000000", "0"),
-        ("0.0", "0"),
+        ("0.000000", "0.0"),
+        ("0.0", "0.0"),
         ("1.0", "1"),
         ("0.123000", "0.123"),
         ("0.000123000", "0.000123"),
-        ("0.0", "0"),
-        ("0", "0"),
+        ("0.0", "0.0"),
+        ("0", "0.0"),
         ("1", "1"),
         ("1.0", "1"),
         ("1.01", "1.01"),
         ("1.010", "1.01"),
-        ("0000.0000", "0"),
+        ("0000.0000", "0.0"),
         ("0.10101010", "0.1010101"),
         ("50000000000", "50000000000"),
-        ("99900000000000000123", "99900000000000000123"),
-        ("99900000000000000123.456000", "99900000000000000123.456"),
+        ("99900000000000000123", "99900000000000000000"),
+        ("99900000000000000123.456000", "99900000000000000000"),
```

Comment:
While it's probably fine that `FloatObject` is no longer capable of such exact precision at the extreme ends, if this were to be merged, then this change should be well communicated to users to avoid any surprises.

Reply:
Reviewing code of Sumatra and Pdf.js, 64-bits floating points are used for internal storage. The rounding will be done. This should be transparent to users. I propose to add a comment in changelog and depreciation page